tqec/detectors/measurement_map.py

```diff
@@ -122,50 +109,25 @@ def _get_global_measurement_index(
             - global_measurement_index is an integer representing the index of the next measurement that
               will be encountered. It is part of the return API to simplify the recursion, and should not
               be useful for the external caller.
-        :raises AssertionError: see pre-conditions in docstring. A failed pre-condition is not guaranteed to
-            raise an exception, so be careful about the pre-conditions here.
+        :raises AssertionError: if the provided cirq.AbstractCircuit instance contains an instance of
+            cirq.CircuitOperation.
         """
         global_measurement_indices: list[dict[cirq.Qid, int]] = []
         global_measurement_index: int = _measurement_offset
         for moment in circuit:
             global_measurement_indices.append(dict())
-            moment_contains_measurement: bool = False
-            moment_contains_CircuitOperation: bool = False
             for op in moment.operations:
+                # Note that cirq.CircuitOperation instances should never be encountered here because
+                # they should all have been flattened at the MeasurementMap creation.
+                assert not isinstance(
+                    op.untagged, cirq.CircuitOperation
+                ), "cirq.CircuitOperation should be flattened before calling this method."
```

Comment:
To make this a little more explicit, you could create a `FlattenedCircuitOperation` class that is the desired input type.